fixing sub-property externals case for validation only

fresheneesz opened the pull request:
(eg {externals:{external:['a','b']}} ). Note that the actual handling of sub-properties still isn't working.

<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->

**What kind of change does this PR introduce?**

Bugfix

**Did you add tests for your changes?**

Yes

**If relevant, link to documentation update:**

N/A

**Summary**

fixes #5631

**Does this PR introduce a breaking change?**

No

**Other information**

This shouldn't be merged in without properly implementing externals sub-properties.

sokra commented on schemas/webpackOptionsSchema.json:
```diff
@@ -823,7 +823,8 @@
     },
     "externals": {
-      "$ref": "#/definitions/externals"
+      "type": "object",
```
``` js
externals: /xyz/
externals: ["a", "b"]
externals: "xyz"
```

These are also valid options.

fresheneesz replied:
Ah ok. I'll correct that

fresheneesz replied:
Actually, what does the second two mean? I don't see those forms documented or tested:

```
externals: ["a", "b"]
externals: "xyz"
```

Especially with the array, its ambiguous to me whether its defining that both "a" and "b" are external modules, or whether it means a sub property corresponding to `root["a"]["b"]`.

fresheneesz replied:
Ok no I got it. `["a","b"]` is defining both of those as external modules. Will fix it.

fresheneesz commented:
Ok updated the schema to support the following forms:
```
externals: /x/
externals: "x"
externals: function(){}
externals: ["a","b",/c/, function(){}, {}]
externals: {
 a: "a",
 "b/c": ["b","c"],
 d: {
   umd: "d",
   root: ["_","d"]
 }
}
```

fresheneesz commented:
@webpack-bot Alright then, didn't realize that mattered.

sokra commented on schemas/webpackOptionsSchema.json:
```diff
@@ -75,6 +75,20 @@
+      ],
+      "description": "Represents a replacement object for an external modules (configured by `externals`)"
+    },
     "externals": {
```
external**s** is now unused?

fresheneesz replied:
Oops, that wasn't intentional

sokra commented on schemas/webpackOptionsSchema.json:
```diff
@@ -108,7 +112,17 @@
         {
           "items": {
-            "$ref": "#/definitions/externals"
+            "anyOf": [
```
now

``` js
externals: [
  /abc/,
  () => {}
]
```

is no longer allowed?

sokra commented on schemas/webpackOptionsSchema.json:
```diff
@@ -83,17 +97,7 @@
-                "type": "boolean"
-              }
-            ]
+            "$ref": "#/definitions/external"
```
``` js
externals: {
  abc: {
    root: "Abc",
    commonjs: "abc"
  }
}
```

is no longer allowed?

fresheneesz commented:
@sokra Ok I added some additional test cases and fixed the webpackOptionsSchema so that it doesn't seem to be getting validation errors. I'm surprised, tho, that the tests seem to be failing *after* validation. Maybe I wrote the tests wrong?

vankop commented:
Thanks for your contribution. I am closing PR because it is too stale..

fresheneesz commented:
Sorry, it seems I lost track of this. Is this PR still useful? Should I try to fix the issues?
